streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -755,7 +755,14 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester
         assert(outputBuffer.asScala.flatten.toSet === expectedOutput.toSet)
       }
     } finally {
-      Utils.deleteRecursively(testDir)
+      try {
+        // As the driver shuts down in the middle of processing above, `testDir` is not closed
+        // correctly which causes the test failure on Windows.
+        Utils.deleteRecursively(testDir)
+      } catch {
+        case e: IOException if Utils.isWindows =>
+          logWarning(e.getMessage)
+      }
```

Comment:
Here, it seems it shuts down in the middle of processing for testing. So, it looks ending up with not closing it.